public/atletacompeticao/acao.php

```diff
@@ -0,0 +1,273 @@
+<?php
+
+require_once(__DIR__.'/../../vendor/autoload.php');
+
+use App\Util\Http\Request;
+use App\Util\Http\Response;
+use App\Util\Database\Connection;
+use App\Tecnico\Tecnico;
+use App\Tecnico\Atleta\Atleta;
+use App\Tecnico\Atleta\AtletaCompeticao\AtletaCompeticaoRepository;
+use App\Tecnico\Atleta\AtletaCompeticao\AtletaCompeticao;
+use App\Tecnico\Atleta\AtletaCompeticao\AtletaCompeticaoCategoria;
+use App\Tecnico\Atleta\AtletaCompeticao\AtletaCompeticaoCategoriaRepository;
+use App\Tecnico\Atleta\AtletaRepository;
+use App\Tecnico\Atleta\AtletaCompeticao\AtletaCompeticaoDuplaRepository;
+use App\Tecnico\Atleta\AtletaCompeticao\AtletaCompeticaoDupla;
+use App\Categorias\Categoria;
+use App\Categorias\CategoriaRepository;
+use App\Tecnico\Atleta\Sexo;
+use App\Util\Http\HttpStatus;
+use App\Util\Services\UploadImagemService\UploadImagemService;
+use App\Util\Exceptions\ValidatorException;
+use App\Util\General\Dates;
+
+try {
+    atletaCompeticaoController()->enviar();
+} catch (Exception $e) {
+    Response::erroException($e)->enviar();
+}
+
+
+function atletaCompeticaoController(): Response
+{
+    $acao = $_POST['acao'] ?? 'Ação não informada';
+    $pdo  = Connection::getInstance();
+    return match ($acao) {
+        'cadastrar' => realizarCadastro($pdo),
+        default => Response::erro('Ação inválida', ['acao' => $acao]),
+    };
+}
+
+function realizarCadastro($pdo) : Response
+{
+
+    if($_POST['userChoice']){
+        return $_POST['userChoice'] == 1 ? realizarCadastroAtletaSelecionado($pdo) : realizarCadastroComNovoAtleta($pdo);
+    }
+
+    return Response::erro('Não foi possível identificar a opção de cadastro selecionada. Por favor escolha entre "Selecionar atleta cadastrado" ou "Cadastrar novo atleta".');
+}
+
+function realizarCadastroAtletaSelecionado($pdo): Response
+{
+    try{
+        $pdo->beginTransaction();
+        $retorno = cadastrarAtletaCompeticao($pdo, new Atleta());
+
+        $pdo->commit();
```

Comment:
Pode ser que um `rollback` já tenha sido feito quando o código chega aqui, causando um erro de "There is no active transaction". De fato, sempre que o `$retorno` é uma `Response` com erro (por exemplo quando o atleta não pode jogar na categoria selecionada pela idade), é feito antes um `rollback`:
![image](https://github.com/MateuxDotDev/badminton-manager/assets/48414967/ffcff2ca-c4e8-48b5-b6a3-6fed3b39e256)
A solução mais direta seria verificar se o pdo está dentro de uma transaction antes de fazer o commit, mas acho que isso não lida diretamente com a raíz do problema. O verdadeiro problema é que deveríamos fazer todas as validações (verificar se vieram categorias e duplas na request, verificar se o atleta pode jogar nas categorias que selecionou, verificar se atleta já não está cadastrado na competição etc. etc.) **antes** de iniciar a transaction. Aí nem teríamos que lidar com esse problema de rollback seguido de commit.